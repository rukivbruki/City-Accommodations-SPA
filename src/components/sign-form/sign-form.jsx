import React from "react";
import {connect} from "react-redux";
import PropTypes from "prop-types";
import {Operation} from "../../reducer/user/user";
import {history} from "../../index";
import {PageAddress} from "../../constants";

const SignForm = (props) => {
  const {onInputChange, sendAuthData, email, password} = props;

  const handleSubmitForm = (evt) => {
    evt.preventDefault();

    sendAuthData({
      email,
      password
    });
    history.push(PageAddress.MAIN);
  };

  return (
    <form
      onSubmit={handleSubmitForm}
      className="login__form form"
      action="#"
      method="post"
    >
      <div className="login__input-wrapper form__input-wrapper">
        <label className="visually-hidden">E-mail</label>
        <input
          onChange={onInputChange}
          value={email || ``}
          className="login__input form__input"
          name="email"
          placeholder="Email"
          type="email"
          required
        />
      </div>
      <div className="login__input-wrapper form__input-wrapper">
        <label className="visually-hidden">Password</label>
        <input
          onChange={onInputChange}
          className="login__input form__input"
          value={password || ``}
          name="password"
          placeholder="Password"
          type="password"
          required
        />
      </div>
      <button className="login__submit form__submit button" type="submit">Sign in</button>
    </form>
  );
};

const mapDispatchToProps = (dispatch) => ({
  sendAuthData: (data) => dispatch(Operation.sendAuthData(data))
});

SignForm.propTypes = {
  onInputChange: PropTypes.func.isRequired,
  email: PropTypes.string,
  password: PropTypes.string,
  sendAuthData: PropTypes.func.isRequired
};

export {SignForm};
export default connect(null, mapDispatchToProps)(SignForm);
