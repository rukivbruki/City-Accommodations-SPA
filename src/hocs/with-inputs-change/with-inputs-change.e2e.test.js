import React from "react";
import Enzyme, {mount} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from 'enzyme-to-json';
import withInputsChange from "./with-inputs-change";
import {SignForm} from "../../components/sign-form/sign-form";

jest.mock(`../../index.js`, () => jest.fn().mockReturnValue(null));
jest.mock(`../../constants.js`, () => jest.fn().mockReturnValue(null));

Enzyme.configure({adapter: new Adapter()});

describe(`WithSignForm wrapper work correct`, () => {
  const FormWrapped = withInputsChange(SignForm);

  const wrappedComponent = mount(
      <FormWrapped
        sendAuthData={jest.fn()}
      />
  );

  const emailInput = wrappedComponent.find(`.login__input`).first();
  const passwordInput = wrappedComponent.find(`.login__input`).at(1);

  it(`should render correctly`, () => {
    expect(toJson(wrappedComponent)).toMatchSnapshot();
  });

  it(`change state email on input change`, () => {
    emailInput.simulate(`change`, {target: {value: `shramko.web@yahoo.com`, name: `email`}});

    expect(wrappedComponent.state().email).toEqual(`shramko.web@yahoo.com`);
  });

  it(`change state password on input change`, () => {
    passwordInput.simulate(`change`, {target: {value: `qwerty`, name: `password`}});

    expect(wrappedComponent.state().password).toEqual(`qwerty`);
  });
});
