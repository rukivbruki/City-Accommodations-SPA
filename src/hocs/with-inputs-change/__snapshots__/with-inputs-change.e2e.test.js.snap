// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithSignForm wrapper work correct should render correctly 1`] = `
<WithInputsChange
  sendAuthData={[MockFunction]}
>
  <SignForm
    onInputChange={[Function]}
    sendAuthData={[MockFunction]}
  >
    <form
      action="#"
      className="login__form form"
      method="post"
      onSubmit={[Function]}
    >
      <div
        className="login__input-wrapper form__input-wrapper"
      >
        <label
          className="visually-hidden"
        >
          E-mail
        </label>
        <input
          className="login__input form__input"
          name="email"
          onChange={[Function]}
          placeholder="Email"
          required={true}
          type="email"
          value=""
        />
      </div>
      <div
        className="login__input-wrapper form__input-wrapper"
      >
        <label
          className="visually-hidden"
        >
          Password
        </label>
        <input
          className="login__input form__input"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          required={true}
          type="password"
          value=""
        />
      </div>
      <button
        className="login__submit form__submit button"
        type="submit"
      >
        Sign in
      </button>
    </form>
  </SignForm>
</WithInputsChange>
`;
